test/auth_cluster/test_raft_service_levels.py

```diff
@@ -99,5 +99,34 @@ async def test_service_levels_upgrade(request, manager: ManagerClient):
     result_with_sl_v2 = await cql.run_async(f"SELECT service_level FROM system.service_levels_v2")
     assert set([sl.service_level for sl in result_with_sl_v2]) == set(sls + [sl_v2])
 
-    
+@pytest.mark.asyncio
+async def test_service_levels_work_during_recovery(manager: ManagerClient):
+    servers = await manager.servers_add(3)
+
+    logging.info("Waiting until driver connects to every server")
+    cql = manager.get_cql()
+    hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+
+    logging.info("Creating a bunch of service levels")
+    sls = ["sl" + unique_name() for _ in range(10)]
+    for sl in sls:
+        await cql.run_async(f"CREATE SERVICE LEVEL {sl}")
+
+    logging.info("Validating service levels were created in v2 table")
+    result = await cql.run_async("SELECT service_level FROM system.service_levels_v2")
+    for sl in result:
+        assert sl.service_level in sls
+
+    logging.info(f"Restarting hosts {hosts} in recovery mode")
+    await asyncio.gather(*(enter_recovery_state(cql, h) for h in hosts))
+    for srv in servers:
+        await restart(manager, srv)
+    cql = await reconnect_driver(manager)
+
+    logging.info("Cluster restarted, waiting until driver reconnects to every server")
+    hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+
+    logging.info("Checking service levels can be read and v2 table is used")
+    recovery_result = await cql.run_async("LIST ALL SERVICE LEVELS")
+    assert set([sl.service_level for sl in recovery_result]) == set(sls)
```

Comment:
It would be great if we had a test (we could extend this one) which checked that service levels also work after exiting the recovery mode and finishing the recovery procedure.

Reply:
added